src/scripts/main.js

```diff
@@ -1,3 +1,25 @@
 'use strict';
 
-// write your code here
+const table = document.querySelector('table');
+
+// Перевіряємо, чи існує таблиця
+if (table) {
+  const rows = Array.from(table.rows);
+
+  // Створюємо DocumentFragment для уникнення переробок DOM
+  const fragment = document.createDocumentFragment();
```

Comment:
You created a document fragment but you're not using it anywhere. The purpose of a DocumentFragment is to allow us to perform operations on child nodes without triggering a reflow of the document. You should append your cloned cells to the fragment, and then append the fragment to the table at the end.